src/lib/cfa/ntp_conf.rb

```diff
@@ -0,0 +1,387 @@
+    #
+    # This class is an AugeasElement wrapper. A Record
+    # has a value, and could also contains a comment and
+    # options. Its contain is stored in an AugeasElement.
+    #
+    # Each ntp entry type has different interpretation
+    # for its options in the AugeasElement. For each one,
+    # a Record subclass is created.
+    class Record
+      # Creates the corresponding subclass object according
+      # to its AugeasElement key.
+      # @param [String] key
+      def self.new_from_augeas(augeas_entry)
+        record = record_class(augeas_entry[:key]).new
+        record.augeas = augeas_entry
+        record
+      end
+
+      # Returns the corresponding subclass
+      # @param [string] key
+      def self.record_class(key)
+        entry_type = key.gsub("[]", "")
+        record_class = ["::CFA::NtpConf::", entry_type.capitalize, "Record"].join
+        Kernel.const_get(record_class)
+      end
+
+      def initialize(key = nil)
+        @augeas = { key: key, value: nil }
+      end
+
+      attr_accessor :augeas
+
+      def value
+        tree_value? ? tree_value.value : @augeas[:value]
+      end
+
+      def value=(value)
+        if tree_value?
+          tree_value.value = value
+        else
+          @augeas[:value] = value
+        end
+      end
+
+      def comment
+        return nil unless tree_value?
+        tree_value.tree["#comment"]
+      end
+
+      def comment=(comment)
+        create_tree_value unless tree_value?
+        if comment.to_s == ""
+          tree_value.tree.delete("#comment")
+        else
+          tree_value.tree["#comment"] = comment
+        end
+      end
+
+      def ==(other)
+        other.class == self.class &&
+          other.augeas == augeas
+      end
+
+      alias_method :eql?, :==
+
+      def type
+        augeas[:key].gsub("[]", "")
+      end
+
+      # Returns an String representing the options
+      def raw_options
+        options.join(" ")
+      end
+
+      # Sets options from a String
+      def raw_options=(raw_options)
+        self.options = split_raw_options(raw_options)
+      end
+
+    private
+
+      def tree_value?
+        @augeas[:value].is_a?(AugeasTreeValue)
+      end
+
+      def tree_value
+        tree_value? ? @augeas[:value] : nil
+      end
+
+      def create_tree_value
+        @augeas[:value] = AugeasTreeValue.new(AugeasTree.new, @augeas[:value])
+      end
+
+      def split_raw_options(raw_options)
+        raw_options.to_s.strip.gsub(/\s+/, " ").split(" ")
+      end
+
+      def augeas_options
+        tree_value.tree.select(options_matcher)
+      end
+
+      def options_matcher
+        Matcher.new { |k, _v| !k.include?("#comment") }
+      end
+    end
+
+    # class to represent a ntp command record entry. There is a
+    # subclass for server, peer, broadcast and broacastclient.
+    class CommandRecord < Record
+      def options
+        return [] unless tree_value?
+        augeas_options.map { |option| option[:key] }
```

Comment:
I think it is correct in ruby. In ruby, access control is slighly different to other languages as C++. In fact, private means it only can be passed to self.